File: Golang/Minimum_Size_Subarray_Sum.go
func minSubArrayLen(target int, nums []int) int {
	left, sum := 0, 0
	minLength := len(nums) + 1

	for right := 0; right < len(nums); right++ {
		sum += nums[right]

		for sum >= target {
			if right-left+1 < minLength {
				minLength = right - left + 1
			}
			sum -= nums[left]
			left++
		}
	}

	if minLength > len(nums) {
		return 0
	}
	return minLength
}

File: Kotlin/Minimum_Size_Subarray_Sum.kt
class Solution {
    fun minSubArrayLen(target: Int, nums: IntArray): Int {
        var left = 0
        var sum = 0
        var minLength = Int.MAX_VALUE
        
        for (i in nums.indices) {
            sum += nums[i]
            while (sum >= target) {
                minLength = minOf(minLength, i - left + 1)
                sum -= nums[left]
                left++
            }
        }
        return if (minLength == Int.MAX_VALUE) 0 else minLength
    }
}
